File: commons/src/main/java/com/twh/commons/ServerMetaData.java
package com.twh.commons;

import com.google.common.base.Charsets;
import lombok.*;

/**
 * @author tanwenhai@bilibili.com
 */
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class ServerMetaData {
    private ServerType serverType;

    private ServerStatus serverStatus;

    private String ip;

    private int port;

    public byte[] toJsonByteArray() {
        return JsonUtils.obj2str(this).getBytes(Charsets.UTF_8);
    }

    public ServerMetaData from(byte[] bytes) {
        return JsonUtils.json2pojo(new String(bytes, Charsets.UTF_8), ServerMetaData.class);
    }
}

File: game-core/src/main/kotlin/com/twh/core/ServerMetaData.kt
package com.twh.core

import com.google.common.base.Charsets
import com.twh.commons.JsonUtils

class ServerMetaData {
    private val serverType: ServerType? = null

    private val serverStatus: ServerStatus? = null

    private val ip: String? = null

    private val port: Int = 0

    fun toJsonByteArray(): ByteArray {
        return JsonUtils.obj2str(this)!!.toByteArray(Charsets.UTF_8)
    }

    fun from(bytes: ByteArray): ServerMetaData? {
        return JsonUtils.json2pojo(String(bytes, Charsets.UTF_8), ServerMetaData::class.java)
    }
}